test/e2e/vmanomaly_test.go

```diff
@@ -0,0 +1,565 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/net/context"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	vmv1 "github.com/VictoriaMetrics/operator/api/operator/v1"
+	vmv1beta1 "github.com/VictoriaMetrics/operator/api/operator/v1beta1"
+	"github.com/VictoriaMetrics/operator/internal/controller/operator/factory/finalize"
+)
+
+const anomalyConfig = `
+reader:
+  class: vm
+  datasource_url: https://play.victoriametrics.com/
+  query_range_path: /api/v1/query_range
+  sampling_period: 10s
+  tenant_id: "0"
+  queries:
+    test1:
+      expr: node_arp_entries
+    test2:
+      expr: node_arp_entries
+    test3:
+      expr: node_arp_entries
+    test4:
+      expr: node_arp_entries
+writer:
+  class: noop
+models:
+  model1:
+    class: 'zscore'
+    z_threshold: 2.5
+  model2:
+    class: 'auto'
+    tuned_class_name: 'zscore'
+schedulers:
+  scheduler1:
+    class: "scheduler.periodic.PeriodicScheduler"
+    infer_every: "1m"
+    fit_every: "2m"
+    fit_window: "1h"
+  scheduler2:
+    class: "scheduler.periodic.PeriodicScheduler"
+    infer_every: "2m"
+    fit_every: "4m"
+    fit_window: "1h"
+`
+
+//nolint:dupl,lll
+var _ = Describe("test vmanomaly Controller", Label("enterprise"), Ordered, func() {
+	ctx := context.Background()
+	namespace := "default"
+	BeforeAll(func() {
+		licenseKey := os.Getenv("LICENSE_KEY")
```

Comment:
How contributor outside of VictoriaMetrics team should run vmanomaly test? 

 I suggest to add a fast path. If `licenseKey` is empty, print warning and exit earlier without any tests.

Reply:
done